src/backend/app.js

```diff
@@ -39,11 +39,11 @@ app.use(function(req, res, next) {
 app.use(function(err, req, res, next) {
```

Comment:
if the 404 handler above already has an error you shouldn't overwrite it with 404. browse through this and see if you can rewrite to handle 404s and errors thrown by your controllers: https://expressjs.com/en/guide/writing-middleware.html